prometheus-exporters-formula/scripts/migrate_formula_data.py

```diff
@@ -53,8 +56,19 @@ def migrate(self):
                 self.migrate_from_version_05()
             if schema_version <= 1:
                 self.migrate_from_version_1()
-            with open(self.filepath, 'w') as output:
-                json.dump(self.data, output)
+
+    def dump(self):
+        with open(self.filepath, 'w') as output:
+            json.dump(self.data, output)
+
+
+def none_value_to_empty_string(data):
+    if type(data) is dict:
```

Comment:
Generally it is recommended to use `isinstance` for type checking and to use abstract types over concrete ones. The "pythonic" way would be to emprace duck-typing, e.g.

```python
try:
    for k, v in data.items():
        ...
except AttributeError:
    pass
```
If you want to check types, I would use
```python
    # at the top of scripts.migrate_formular_data
    from collections.abc import Mapping
    # here
    if isinstance(data, Mapping):
        ...
```

Reply:
Good point. Updated to check for the interface.